apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClientPartial.cs

```diff
@@ -85,10 +89,91 @@ public partial class BigtableServiceApiSettings
+        /// </para>
+        /// </remarks>
+        public uint MaxChannels { get; set; } = 16;
+
+        // TODO: Link to snippet for creating GcpCallInvoker manually from here.
+        /// <summary>
+        /// The number of streams open concurrently in a channel which will trigger a new channel to be opened, as
+        /// long as fewer than <see cref="MaxChannels"/> are currently open.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that channels are managed and shared amongst <see cref="BigtableClient"/> instances using the same
+        /// endpoint and channel options and which use the default credentials. So this property will have no bearing
+        /// across clients using different options. To create a custom grouping of channels to be managed, create a
+        /// <see cref="GcpCallInvoker"/> manually and use the
+        /// <see cref="BigtableClient.Create(CallInvoker, BigtableServiceApiSettings, string)">BigtableClient.Create</see>
+        /// overload taking a <see cref="CallInvoker"/> to create clients from it.
+        /// </para>
+        /// </remarks>
+        public uint PreferredMaxStreamsPerChannel { get; set; } = 5;
+
         partial void OnCopy(BigtableServiceApiSettings existing)
         {
             MutateRowsRetrySettings = existing.MutateRowsRetrySettings;
             ReadRowsRetrySettings = existing.ReadRowsRetrySettings;
+            MaxChannels = existing.MaxChannels;
+            PreferredMaxStreamsPerChannel = existing.PreferredMaxStreamsPerChannel;
+        }
+    }
+
+    /// <summary>
+    /// Extensions for use on <see cref="BigtableServiceApiSettings"/>.
+    /// </summary>
+    public static class BigtableServiceApiSettingsExtensions
+    {
+        /// <summary>
+        /// Creates a collection of <see cref="ChannelOption"/> instances which can be used to create a <see cref="Channel"/>
+        /// or <see cref="GcpCallInvoker"/> pre-configured based on the specified settings (or the default settings if they
+        /// are null).
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that if the options returned are used to create a <see cref="Channel"/>, the <see cref="BigtableServiceApiSettings.MaxChannels"/>
+        /// and <see cref="BigtableServiceApiSettings.PreferredMaxStreamsPerChannel"/> settings values will be ignored.
+        /// </para>
+        /// </remarks>
+        /// <param name="settings">
+        /// The settings with which to create channel options. May be null, in which case the default settings will be used.
+        /// </param>
+        /// <returns>A collection of <see cref="ChannelOption"/> instances.</returns>
+        public static IEnumerable<ChannelOption> CreateChannelOptions(this BigtableServiceApiSettings settings)
+        {
+            var effectiveSettings = settings ?? BigtableServiceApiSettings.GetDefault();
```

Comment:
It's slightly annoying that there's so much work involved on every client creation. It might be worth having a default channel option array which is returned immediately if `settings` is null, given that's the most likely case. That can definitely come in a follow-up PR though.

Reply:
Good idea. I'll add a TODO.